File: practice_java_program/notpad/StringCountingLogic.java
public class StringCountingLogic {

    public static void main(String[] args) {

        String input = "aaabbc";     //output="a3b2c1"
        StringBuilder sb = new StringBuilder();

        int count = 1;

        for (int i = 1; i <= input.length(); i++) {

            if (i < input.length() && input.charAt(i) == input.charAt(i - 1)) {
                count++;
            } else {
                sb.append(input.charAt(i - 1)).append(count);
                count = 1;
            }
        }

        System.out.println(sb.toString());
    }
}

File: python_oriented_interview/notpad/StringCountingLogic.py
input_str = "aaabbc" # output="a3b2c1"
sb = []
count = 1

for i in range(1, len(input_str) + 1):
    if i < len(input_str) and input_str[i] == input_str[i - 1]:
        count += 1
    else:
        sb.append(f"{input_str[i - 1]}{count}")
        count = 1

print("".join(sb))
